File: crates/cli/src/templates/js-astgrep/tests/fixtures/input.rs
// println! macros will be matched by pattern!
// click diff tab to see rewrite and import addition.

fn main() {
    println!("Starting application");

    let user_id = 42;
    println!("User ID: {}", user_id);

    if let Some(config) = load_config() {
        println!("Config loaded: {:?}", config);
    } else {
        println!("Failed to load config");
    }

    // This is a comment with println!("not matched")
    let debug_msg = "println!(\"also not matched\")";

    match process_data() {
        Ok(result) => {
            println!("Processing completed successfully: {:?}", result);
        }
        Err(e) => {
            println!("Error processing data: {}", e);
        }
    }
}

fn load_config() -> Option<String> {
    Some("config".to_string())
}

fn process_data() -> Result<String, String> {
    Ok("data".to_string())
}

File: crates/cli/src/templates/js-astgrep/tests/fixtures/input.cs
// Console.WriteLine() will be matched by pattern!
// click diff tab to see rewrite.

using System;

namespace MyApp
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Hello, World!");
            Console.Write("Debug message: ");
            Console.WriteLine($"User ID: {userId}");
            
            if (isDebugMode)
            {
                Console.WriteLine("Application started successfully");
                Console.Write("Processing...");
            }
            
            // This won't be matched
            var message = "Console.WriteLine(\"not me\")";
            SomeOtherClass.WriteLine("also not matched");
        }
        
        private static void ProcessData()
        {
            Console.WriteLine("Processing data");
            Console.Write("Progress: 50%");
        }
    }
}